hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AbstractFSWAL.java

```diff
@@ -1001,8 +1001,13 @@ protected final boolean append(W writer, FSWALEntry entry) throws IOException {
     doAppend(writer, entry);
     assert highestUnsyncedTxid < entry.getTxid();
     highestUnsyncedTxid = entry.getTxid();
-    sequenceIdAccounting.update(encodedRegionName, entry.getFamilyNames(), regionSequenceId,
-      entry.isInMemStore());
+    if (entry.isCloseRegion()) {
+      // let's clean all the records of this region
+      sequenceIdAccounting.onRegionClose(encodedRegionName);
+    } else {
+      sequenceIdAccounting.update(encodedRegionName, entry.getFamilyNames(), regionSequenceId,
+        entry.isInMemStore());
```

Comment:
So, this flag.... isn't it false when not data? Could we use this instead of the appendData and appendMarker?

Reply:
We have other markers such as open, flush, compaction, etc.

Reply:
This does not make sense... You do not want the closeRegion flag but still want the inMemstore flag? At least with appendData and appendMarker, we could remove at least one of the parameters in the methods of the WAL interface...

Reply:
The isInMemStore flag is there currently? No? I'm not advocating adding anything not already present.

Looking at this patch, the new methods appendData and appendMarker do not seem to buy us much. They do not go deep enough down into AbstractFSWAL.